compiler/ballerina-lang/src/main/java/io/ballerina/projects/JBallerinaBackend.java

```diff
@@ -108,6 +127,13 @@ public class JBallerinaBackend extends CompilerBackend {
     private DiagnosticResult diagnosticResult;
     private boolean codeGenCompleted;
     private final List<JarConflict> conflictedJars;
+    private final HashMap<String, ByteArrayOutputStream> optimizedJarStreams = new HashMap<>();
```

Comment:
Alway use the interface type when defining variables in java. Change in other places as well.
```suggestion
    private final Map<String, ByteArrayOutputStream> optimizedJarStreams = new HashMap<>();
```

Reply:
Fixed in aa8254a61e2f3d88c1fdd944ffe5a5b257ede583.